### PR TITLE
tests: add Windows guest snapshot test

### PR DESCRIPTION
**Pull request by liuw**
Signed-off-by: Wei Liu <liuwe@microsoft.com>

### CODE REVIEW

File: scripts/run_integration_tests_windows.sh [hunk 1]
```diff
@@ -16,7 +16,9 @@ strip target/$BUILD_TARGET/release/cloud-hypervisor
 
 export RUST_BACKTRACE=1
 
-time cargo test --features "integration_tests" "tests::windows::$@"
+# Only run with 1 thread to avoid tests interfering with one another because
+# Windows has a static IP configured
+time cargo test --features "integration_tests" "tests::windows::$@" -- --test-threads=1
```
Comment (sboeuf) [inline scripts/run_integration_tests_windows.sh:21]: Why do we need to serialize the tests here? Is it because of the VM's networking?

**Review by sboeuf — COMMENTED** [review]
@liuw looks good, but I think it's not sustainable in the long term if we keep adding tests and we use only one thread. We'd need something like cloud-init to provision the VM with the right IP address. Or maybe we could send some commands to SAC from STDIN in order to setup the guest IP address.
/cc @rbradford

File: scripts/run_integration_tests_windows.sh (hunk 1 above)
Comment (liuw) [reply scripts/run_integration_tests_windows.sh:21]: Yes. The comment says that already.

Comment (sboeuf) [reply scripts/run_integration_tests_windows.sh:21]: I'm sorry I completely missed the comment :facepalm:

**Review by rbradford — APPROVED** [review]

**Comment by rbradford** [comment]
I think we need to work out how to do more tests efficiently (in terms of IP address allocation) but we shouldn't hold this up.

**Comment by sboeuf** [comment]
@liuw @rbradford I've just opened https://github.com/cloud-hypervisor/cloud-hypervisor/issues/1891
Let's merge this PR now.

**Pull request merged**